fish_prompt.fish

```diff
@@ -81,9 +81,8 @@ function fish_prompt
     end
 
     # Check if there is an upstream configured
-    set -l has_upstream (command git rev-parse --abbrev-ref @'{u}' ^/dev/null)
-
-    if test -n "$has_upstream"
+    command git rev-parse --abbrev-ref '@{upstream}' >/dev/null ^&1
+    if test $status -eq 0
```

Comment:
In term of business domain it's more explicit to test if we have an upstream than checking the exit status (which is fish/shell-related).